### ISSUE TITLE
jest-emotion No support for CSS pseudo-elements and pseudo-classes

### ISSUE THREAD

**Issue by Tokery** [emotion-js/emotion#1208]
- `jest-emotion` version: 10.0.6

**Relevant code:**

I am looking to create tests that can add modifiers to the class selector, like seen below. (From jest styled components)
```javascript
const Button = styled.button`
  @media (max-width: 640px) {
    &:hover {
      color: red;
    }
  }
`

test('it works', () => {
  const tree = renderer.create(<Button />).toJSON()
  expect(tree).toHaveStyleRule('color', 'red', {
    media: '(max-width:640px)',
    modifier: ':hover',
  })
})
```

As far as I'm aware, this doesn't yet exist.

**Comment by Werter12** [comment]
@Tokery Actually, the PR with this functionality was merged https://github.com/emotion-js/emotion/pull/1110. I am going to add documentation for it soon. So, you can use the following syntax:
`
  expect(tree).toHaveStyleRule('color', 'red', {
    media: '(max-width:640px)',
    target: ':hover',
  })
`

**Comment by Tokery** [comment]
Ah, I see. Thanks for the info.

**Comment by stephenh** [comment]
I know this is a closed issue, but I can't get this to work with a non-trivial selector, i.e.:

```
<div css={ { "&:hover > *": { color: "red" } } }> ... </div>

expect(...).toHaveStyleRule("color", "red", { target: "&:hover > *" }
```

Doesn't work, and I've tried various approaches of putting spaces between/not-between `&` / `:` / `>` / `*`, including/not including `&`, etc., and can't seem to get anything to work.

I set a debug point inside of `toMatchStyleRule` and it really looks like the `> *` is not coming back at all for the `target` to later pick it up.

@Werter12 apologies for the ~year later lazy ask on a closed PR, but the docs are still light on this more esoteric selectors; are there any obvious ideas I'm missing? Thanks!

**Comment by Andarist** [comment]
@Werter12 if you set up a simple repro for the problem I would be able to take a quick look at it (as in - debug it and assess if you have passed arguments incorrectly or if maybe we should change something in Emotion's `toMatchStyleRule`)

**Comment by Andarist** [comment]
@stephenh  if you set up a simple repro for the problem I would be able to take a quick look at it (as in - debug it and assess if you have passed arguments incorrectly or if maybe we should change something in Emotion's `toMatchStyleRule`)

**Comment by stephenh** [comment]
@Andarist I played with this a bit and figured out it does basically work, i.e. for this:

```
        css={{
          "&:hover": { marginBottom: "16px", ...Css.pb3.$ },
          "& > div:focus": Css.pb4.$,
          "& > div + div": Css.pt4.$,
        }}
```

The following assertions work:

```
    expect(div).toHaveStyleRule("margin-bottom", "16px", { target: ":hover" });
    expect(div).toHaveStyleRule("padding-bottom", "32px", {
      target: "div:focus",
    });
    expect(div).toHaveStyleRule("padding-top", "32px", {
      target: "div",
    });
```

So, `:hover` is straight forward--just drop the `&`.

For `& > div:focus`, drop the `&` and `>` and it's `div:focus`.

For `& > div + div`, drop the `&` and `>` and `+` and the combine (kinda weird) the `div`s into just one `div` so it's just `div`.

I cheated a little bit on the repro and made a throw-away branch in an existing project that already had emotion / jest-emotion setup:

https://github.com/homebound-team/truss/blob/emotion-style-test/integration-test/Css.emotion.test.tsx#L54

So, I think I'm technically unblocked b/c, with enough attempts, I can basically stripe characters until finding the right selector, but it's not incredibly intuitive. Would be great to tweak this somehow to perhaps leave in some more of the characters? Or maybe just all of them? Fwiw that would have been the most intuitive / what I tried first, i.e. typing the `target: ...` to be exactly like the selector I passed to `css` (i.e. including `&` and including whitespace).

Thanks!
